wsd/COOLWSD.cpp

```diff
@@ -1198,9 +1201,12 @@ class RemoteConfigPoll : public RemoteJSONPoll
 
         fetchRemoteFontConfig(newAppConfig, remoteJson);
 
-        AutoPtr<AppConfigMap> newConfig(new AppConfigMap(newAppConfig));
-        _conf.addWriteable(newConfig, PRIO_JSON);
-        newConfig->clear();
+        // delete old remote config layer before applying the new layer
+        if (!persistConfig.isNull())
+            persistConfig->clear();
+
+        persistConfig = new AppConfigMap(newAppConfig);
+        _conf.addWriteable(persistConfig, PRIO_JSON);
```

Comment:
Are we sure this fixes the leak? It's not obvious to me what happens to the `AppConfigMap` instances we are adding in `_conf` by calling `addWriteable`.

It seems that `_conf` is taking ownership of the `AutoPtr`, so how is this different from the old solution, other than using a member variable instead of a local one?

Shouldn't we instead simply modify the `persistConfig` instance, instead of creating a new one? That is, we create it and add to `_conf` and after that we simply update it. So we never create more than one `AppConfigMap`. Wouldn't that fix the leak and make things simpler?